lib/client.js

```diff
@@ -179,6 +158,27 @@ api._pollServer = function(options, callback) {
   pollServer(options, callback);
 };
 
+function insertClient(client, callback) {
+  var now = Date.now();
+  var meta = {};
+  meta.created = now;
+  meta.updated = now;
+  meta.recentPollErrorCount = 0;
+  var record = {
+    id: database.hash(client.id),
+    meta: meta,
+    client: client
+  };
+  database.collections.messageClient.insert(record, database.writeOptions,
+    function(err, result) {
+    if(err && database.isDuplicateError(err)) {
+      // Don't fail on duplicates
+      return callback();
+    }
+    callback(err);
+  });
```

Comment:
We should just return the record in the callback, not do the duplicate check here. The dup check should be moved to where this function is called above during initialization -- otherwise we may want to make this function public in the future and wouldn't want to mask duplicate errors that the caller may handle in a different way.